packages/rsocket-websocket-client/src/RSocketWebSocketClient.js

```diff
@@ -66,7 +72,7 @@ export default class RSocketWebSocketClient implements DuplexConnection {
         'established.',
     );
     this._setConnectionStatus(CONNECTION_STATUS.CONNECTING);
-    const socket = (this._socket = new WebSocket(this._options.url));
+    const socket = (this._socket = this._wsCreator(this._url, this._options));
```

Comment:
Check if `this._options.wsCreator` is null here (!this._options.wsCreator) and just call `new WebSocket(...)` in this case

Reply:
The constructor now checks if `wsCreator` is null. If it is, it calls `wsCreator(url, wsOptions)`, otherwise it calls `new WebSocket(url)` as it is was before.